src/dawn/SIR/SIR.h

```diff
@@ -282,11 +282,17 @@ struct Value : NonCopyable {
   template <class T>
   explicit Value(T value) : isConstexpr_(false) {
     valueImpl_ = make_unique<ValueImpl<T>>(value);
+    //NOTE: perfect forwarding in this case can be achieved by using 
+    //explicit Value(T value) : isConstexpr_(false) {make_unique<ValueImpl<std::decay_t<T>>>(std::forward<T>(value));}
```

Comment:
`typename std::decay<T>::type` or is there something else I am missing which doesn't work in c++11?